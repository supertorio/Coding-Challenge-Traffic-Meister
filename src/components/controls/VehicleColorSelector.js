import React from 'react';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {List as ImmutableList} from "immutable";

import {getAvailableVehicleColors, getCurrentVehicleColor} from "../../reducers/index-reducer";
import {changeVehicleColorSelection} from '../../actions/meister-form-actions';
import BaseListSelector from "./BaseListSelector";

const VehicleColorSelector = (props) => {
    return (
        <div className="Selector VehicleColorSelector">
            <BaseListSelector label="Vehicle Color"
                              options={props.colors}
                              value={props.selectedColor}
                              onChange={props.changeVehicleColorSelection} />
        </div>
    );
};

VehicleColorSelector.propTypes = {
    colors: PropTypes.instanceOf(ImmutableList).isRequired,
    selectedColor: PropTypes.string,
    changeVehicleColorSelection: PropTypes.func,
};

const mapStateToProps = state => {
    return {
        selectedColor: getCurrentVehicleColor(state),
        colors: getAvailableVehicleColors(state),
    };
};

const mapDispatchToProps = {
    changeVehicleColorSelection,
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(VehicleColorSelector);
