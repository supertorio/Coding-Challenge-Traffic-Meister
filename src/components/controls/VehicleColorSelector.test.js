import React from 'react';
import {VehicleColorSelector} from './VehicleColorSelector';
import {List} from 'immutable';

describe('<VehicleColorSelector />', () => {

    let colors, selectedColor;

    beforeEach(()=> {
        colors = new List(['red', 'green', 'blue']);
        selectedColor = colors.get[0];
    });

    it('should match the snapshot', () => {
        const wrapper = shallow(
            <VehicleColorSelector colors={colors} selectedColor={selectedColor} changeVehicleColorSelection={() => true} />,
        );
        expect(wrapper).toMatchSnapshot();
    });

});
