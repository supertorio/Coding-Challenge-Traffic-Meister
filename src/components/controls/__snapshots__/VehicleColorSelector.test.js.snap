// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VehicleColorSelector /> should match the snapshot 1`] = `ShallowWrapper {}`;
